src/widgets/qownnotesmarkdowntextedit.cpp

```diff
@@ -73,6 +73,15 @@ QOwnNotesMarkdownTextEdit::QOwnNotesMarkdownTextEdit(QWidget *parent)
         onZoom(/*in=*/ false);
     });
 
+    connect(this, &QPlainTextEdit::cursorPositionChanged, this,
+            [this]() {
+        if (!mainWindow) {
+            qWarning() << "No MainWindow! shouldn't happen!";
+            return;
+        }
+        mainWindow->selectNavigationItemAtPosition(textCursor().block().position());
```

Comment:
We still need a setting in the settings dialog for that (default on should be ok), I guess this feature will drive some people crazy. 😉

Reply:
maybe lets wait till someone comes up who wants to turn it off.

Reply:
I may be the first already... I'm not sure yet. 😆 
And as soon as other stuff (like links) are shown in the navigation panel then that feature needs to be turned off..

Reply:
Will be adding the setting tonight.

Reply:
Thank you!